src/main/scala/viper/gobra/frontend/info/implementation/typing/StmtTyping.scala

```diff
@@ -137,6 +137,18 @@ trait StmtTyping extends BaseTyping { this: TypeInfoImpl =>
     case _: PSeq => noMessages
     case _: PEmptyStmt => noMessages
 
+    case n: POutline =>
+      error(n.spec, "Currently, outline statements do not support the pure annotation.", n.spec.isPure) ++ 
+      error(n.spec, "Currently, the old keyword is not supported", (n.spec.posts ++ n.spec.pres).find{exp => 
+        allChildren(exp).find{case _: POld => true case _ => false}.isDefined
+      }.isDefined) ++
+      n.body.collect{case stmt if (allChildren(stmt).find{case _: POld => true case _ => false}.isDefined) => 
+          error(stmt, "Currently, the old keyword is not supported")
+      }.flatten ++
+      n.body.collect{case stmt: PReturn => 
+          error(stmt, "Using return in an outlined block is not allowed")
+      }.flatten
```

Comment:
Shouldn't we also check here that we never take the reference of a variable returned from the enclosing scope? (In this case, variables are passed as arguments to the outline block and thus, they are necessarily exclusive)